packages/pluggableWidgets/accordion-web/src/Accordion.editorConfig.ts

```diff
@@ -105,3 +109,120 @@ export function check(values: AccordionPreviewProps): Problem[] {
+                                            columnSize: "grow",
+                                            children: [
+                                                ...(values.collapsible && values.showIcon === "left"
+                                                    ? [getIconPreview(group, headerTextIconPadding)]
+                                                    : []),
+                                                {
+                                                    type: "Container",
+                                                    padding: group.headerRenderMode === "text" ? 10 : 0,
+                                                    grow: 100,
+                                                    children: [
+                                                        group.headerRenderMode === "text"
+                                                            ? {
+                                                                  type: "Text",
+                                                                  content: group.headerText,
+                                                                  fontSize: headerTextFontSize,
+                                                                  bold: true
+                                                              }
+                                                            : {
+                                                                  type: "DropZone",
+                                                                  property: group.headerContent,
+                                                                  placeholder: `Place header contents for group ${index}`
+                                                              }
+                                                    ]
+                                                },
+                                                ...(values.collapsible && values.showIcon === "right"
+                                                    ? [getIconPreview(group, headerTextIconPadding)]
+                                                    : [])
+                                            ]
+                                        }
+                                    ]
+                                },
+                                {
+                                    type: "Container",
+                                    borders: true,
+                                    children: [
+                                        {
+                                            type: "DropZone",
+                                            property: group.content,
+                                            placeholder: `Place body contents for group ${index}`
+                                        }
+                                    ]
+                                }
+                            ]
+                        }
+                    }
+                ];
```

Comment:
1. I think this part looks pretty fine, but in general I feel like `StructurePreviewProps` constructions become very verbose and hard to follow real quick. You're also using this inside an array function, which doesn't help. So personally I would suggest extracting this entire section to another function as you did with `getIconPreview`: `getGroupPreview(group: GroupsPreviewType, ...): StructurePreviewProps[]`.
2. Then, you don't have to do the map and reduce combination anymore, and you could do something along the likes of
```
children: values.groups.reduce((group, accumulator, index) => {
   const { headerTextFontSize, headerTextIconPadding } = getHeaderStylingPropertiesOrSomething(group);

  return [
    ...accumulator,
    ...getGroupPreview(group, ...)
  ];
}, [])
```

This feels a bit more structured, less everything-in-one-place, and also labels some of the code which is always nice. What do you think?